frontend/asset_mapper.rst

```diff
@@ -0,0 +1,1097 @@
+
+.. code-block:: text
+
+    AssetMapper Paths
+    ------------------
+
+    --------- ------------------
+     Path      Namespace prefix
+    --------- ------------------
+    assets
+
+    Mapped Assets
+    -------------
+
+    ------------------ ----------------------------------------------------
+     Logical Path       Filesystem Path
+    ------------------ ----------------------------------------------------
+     app.js             assets/app.js
+     styles/app.css     assets/styles/app.css
+     images/duck.png    assets/images/duck.png
+
+The "Logical Path" is the path to use when referencing the asset, like
+from a template.
+
+.. _importmaps-javascript:
+
+Importmaps & Writing JavaScript
+-------------------------------
+
+All modern browsers support the JavaScript `import statement`_ and modern
+`ES6`_ features like classes. So this code "just works":
+
+.. code-block:: javascript
+
+    // assets/app.js
+    import Duck from './duck.js';
+
+    const duck = new Duck('Waddles');
+    duck.quack();
+
+.. code-block:: javascript
+
+    // assets/duck.js
+    export default class {
+        constructor(name) {
+            this.name = name;
+        }
+        quack() {
+            console.log(`${this.name} says: Quack!`);
+        }
+    }
+
+Thanks to the ``{{ importmap() }}`` Twig function, which you'll learn all about in
+this section, the ``assets/app.js`` file is loaded & executed by the browser.
+
+.. tip::
+
+    When importing relative files, be sure to include the ``.js`` extension.
+    Unlike in Node, the extension is required in the browser environment.
+
+Importing 3rd Party JavaScript Packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose you want to use an `npm package`_, like `bootstrap`_. Technically,
+this can be done by importing its full URL, like from a CDN:
+
+.. code-block:: javascript
+
+    import { Alert } from 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/+esm';
+
+But yikes! Needing to include that URL is a pain! Instead, we can add
+this to our "importmap" via the ``importmap:require`` command. This command can
+be used to download any `npm package`_:
+
+.. code-block:: terminal
+
+    $ php bin/console importmap:require bootstrap
+
+This adds the ``bootstrap`` package to your ``importmap.php`` file::
+
+    // importmap.php
+    return [
+        // ...
+
+        'bootstrap' => [
+            'url' => 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/+esm',
+        ],
+    ];
+
+Now you can import the ``bootstrap`` package like normal:
+
+.. code-block:: javascript
+
+    import { Alert } from 'bootstrap';
+    // ...
+
+If you want to download the package locally, use the ``--download`` option:
+
+.. code-block:: terminal
+
+    $ php bin/console importmap:require bootstrap --download
+
+This will download the package into an ``assets/vendor/`` directory and update
+the ``importmap.php`` file to point to it. You *should* commit this file to
+your repository.
```

Comment:
Is the vendor directory should also be committed or should be gitignored ? In that case, is the recipe add the folder to gitignore ?

Reply:
Hi! You can't actually commit directories to git - only files. But the meaning here is that you "commit the vendor/ directory and all files inside to git". So the `assets/vendor/` directory should NOT be ignored :)